glusterd: simplify peer state management code

dmantipov opened the pull request:
Since `transition_time` of `glusterd_peer_state_info_t` is not
used anywhere, drop the latter and move `state` field directly
into `glusterd_peerinfo_t`, adjust related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/mgmt/glusterd/src/glusterd-op-sm.c b/xlators/mgmt/glusterd/src/glusterd-op-sm.c
index 2e3630839..9e901bfa8 100644
--- a/xlators/mgmt/glusterd/src/glusterd-op-sm.c
+++ b/xlators/mgmt/glusterd/src/glusterd-op-sm.c
@@ -131,8 +131,8 @@ glusterd_txn_opinfo_dict_fini()
 
 void
 glusterd_txn_opinfo_init(glusterd_op_info_t *opinfo,
-                         glusterd_op_sm_state_t state, int *op,
-                         dict_t *op_ctx, rpcsvc_request_t *req)
+                         glusterd_op_sm_state_t state, int *op, dict_t *op_ctx,
+                         rpcsvc_request_t *req)
 {
     glusterd_conf_t *conf = NULL;
 
@@ -5826,9 +5826,9 @@ glusterd_op_sm_transition_state(glusterd_op_info_t *opinfo,
     conf = THIS->private;
     GF_ASSERT(conf);
 
-    (void)glusterd_sm_tr_log_transition_add(
-        &conf->op_sm_log, opinfo->state, state[event_type].next_state,
-        event_type);
+    (void)glusterd_sm_tr_log_transition_add(&conf->op_sm_log, opinfo->state,
+                                            state[event_type].next_state,
+                                            event_type);
 
     opinfo->state = state[event_type].next_state;
     return 0;
diff --git a/xlators/mgmt/glusterd/src/glusterd-peer-utils.c b/xlators/mgmt/glusterd/src/glusterd-peer-utils.c
index bfa5488a4..b09ca84cd 100644
--- a/xlators/mgmt/glusterd/src/glusterd-peer-utils.c
+++ b/xlators/mgmt/glusterd/src/glusterd-peer-utils.c
@@ -940,9 +940,8 @@ gd_add_peer_detail_to_dict(glusterd_peerinfo_t *peerinfo, dict_t *friends,
     }
 
     keylen = snprintf(key, sizeof(key), "friend%d.state", count);
-    ret = dict_set_strn(
-        friends, key, keylen,
-        glusterd_friend_sm_state_name_get(peerinfo->state));
+    ret = dict_set_strn(friends, key, keylen,
+                        glusterd_friend_sm_state_name_get(peerinfo->state));
     if (ret) {
         gf_smsg(this->name, GF_LOG_ERROR, 0, GD_MSG_DICT_SET_FAILED, "key=%s",
                 key, NULL);
diff --git a/xlators/mgmt/glusterd/src/glusterd-sm.c b/xlators/mgmt/glusterd/src/glusterd-sm.c
index 8eab5e325..85e713507 100644
--- a/xlators/mgmt/glusterd/src/glusterd-sm.c
+++ b/xlators/mgmt/glusterd/src/glusterd-sm.c
@@ -1103,9 +1103,9 @@ glusterd_friend_sm_transition_state(uuid_t peerid, char *peername,
         goto out;
     }
 
-    (void)glusterd_sm_tr_log_transition_add(
-        &peerinfo->sm_log, peerinfo->state, state[event_type].next_state,
-        event_type);
+    (void)glusterd_sm_tr_log_transition_add(&peerinfo->sm_log, peerinfo->state,
+                                            state[event_type].next_state,
+                                            event_type);
 
     uatomic_set(&peerinfo->state, state[event_type].next_state);
 
```

mykaul commented:
LGTM.

dmantipov commented:
/run regression

Commit:
glusterd: simplify peer state management code

Since 'transition_time' of 'glusterd_peer_state_info_t' is not
used anywhere, drop the latter and move 'state' field directly
into 'glusterd_peerinfo_t', adjust related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

dmantipov commented:
/run regression

Commit:
Merge branch 'devel' into 1000-glusterd-simplify-peer-state-management-code

dmantipov commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/bugs/quota/bug-1035576.t

0 test(s) generated core 


7 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/basic/afr/metadata-self-heal.t
./tests/bugs/distribute/issue-2317.t
./tests/bugs/quota/bug-1035576.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2336/

dmantipov commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/georep-basic-dr-tarssh.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/georep-basic-dr-tarssh.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2340/

dmantipov commented:
/run regression

Commit:
Merge branch 'devel' into 1000-glusterd-simplify-peer-state-management-code

dmantipov commented:
Closed, will be replaced by new pull request merged with https://github.com/gluster/glusterfs/pull/2961.

gluster-ant commented:
!!! Couldn't read commit file !!!
